src/navsat_transform.cpp

```diff
@@ -67,52 +68,46 @@ NavSatTransform::NavSatTransform(std::shared_ptr<rclcpp::Node> node) :
   magnetic_declination_(0.0),
   odom_updated_(false),
   publish_gps_(false),
-  tf_buffer_(node->get_clock()),
+  tf_buffer_(this->get_clock()),
   tf_listener_(tf_buffer_),
   transform_good_(false), 
+  transform_timeout_(0ns),
   use_manual_datum_(false), 
   use_odometry_yaw_(false), 
-  utm_broadcaster_(node),
+  utm_broadcaster_(*this),
   utm_odom_tf_yaw_(0.0), 
   utm_zone_(""), 
-  world_frame_id_("odom"), 
+  world_frame_id_("odom"),
   yaw_offset_(0.0),
-  transform_timeout_(0ns), 
   zero_altitude_(false)
 {
   latest_utm_covariance_.resize(POSE_SIZE, POSE_SIZE);
   latest_odom_covariance_.resize(POSE_SIZE, POSE_SIZE);
-  node_ = node;
-}
-
-NavSatTransform::~NavSatTransform() {}
 
-void NavSatTransform::run()
-{
   double frequency = 10.0;
   double delay = 0.0;
   double transform_timeout = 0.0;
 
   // Load the parameters we need
-  magnetic_declination_ = node_->declare_parameter("magnetic_declination_radians", 0.0);
-  yaw_offset_ = node_->declare_parameter("yaw_offset", 0.0);
-  broadcast_utm_transform_ = node_->declare_parameter("broadcast_utm_transform", false);
-  broadcast_utm_transform_as_parent_frame_ = node_->declare_parameter("broadcast_utm_transform_as_parent_frame", false);
-  zero_altitude_ = node_->declare_parameter("zero_altitude", false);
-  publish_gps_ = node_->declare_parameter("publish_filtered_gps", false);
-  use_odometry_yaw_ = node_->declare_parameter("use_odometry_yaw", false);
-  use_manual_datum_ = node_->declare_parameter("wait_for_datum", false);
-  frequency = node_->declare_parameter("frequency", 10.0);
-  delay = node_->declare_parameter("delay", 0.0);
-  transform_timeout = node_->declare_parameter("transform_timeout", 0.0);
+  magnetic_declination_ = this->declare_parameter("magnetic_declination_radians", 0.0);
+  yaw_offset_ = this->declare_parameter("yaw_offset", 0.0);
+  broadcast_utm_transform_ = this->declare_parameter("broadcast_utm_transform", false);
+  broadcast_utm_transform_as_parent_frame_ = this->declare_parameter("broadcast_utm_transform_as_parent_frame", false);
+  zero_altitude_ = this->declare_parameter("zero_altitude", false);
+  publish_gps_ = this->declare_parameter("publish_filtered_gps", true);
+  use_odometry_yaw_ = this->declare_parameter("use_odometry_yaw", false);
+  use_manual_datum_ = this->declare_parameter("wait_for_datum", false);
+  frequency = this->declare_parameter("frequency", 10.0);
```

Comment:
You have frequency defined above, use that variable. Same with delay/timeout.

ei `   frequency = this->declare_parameter("frequency", frequency); `